hermes/bin/src/runtime_extensions/hermes/crypto/state.rs

```diff
@@ -0,0 +1,222 @@
+//! Crypto state
+
+use std::{
+    cmp::Eq,
+    hash::{Hash, Hasher},
+    sync::Arc,
+};
+
+use dashmap::DashMap;
+use ed25519_bip32::XPrv;
+//  std::sync::LazyLock is still unstable
+use once_cell::sync::Lazy;
+use rusty_ulid::Ulid;
+
+/// Map of app name, module ULID, event name, and module execution counter to resource
+/// holder
+type State = DashMap<String, DashMap<Ulid, DashMap<String, DashMap<u64, ResourceHolder>>>>;
+
+/// Wrapper for XPrv to implement Hash used in DashMap
+#[derive(Eq, Clone, PartialEq)]
+struct WrappedXPrv(XPrv);
+
+/// Implemnt Hash for WrappedXPrv
+impl Hash for WrappedXPrv {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.as_ref().hash(state);
+    }
+}
+
+impl WrappedXPrv {
```

Comment:
This should probably be better as:
```rust
impl From<XPrv> for WrappedXPrv {
   ...
}
```
see: https://doc.rust-lang.org/rust-by-example/conversion/from_into.html